### PR TITLE
[objdump] Split out state from ObjdumpOptions

### PR DESCRIPTION
**Pull request by sbc100**
Also, use C++ constructor inheritance to avoid some
boilerplate constructors.

### CODE REVIEW

File: src/binary-reader-objdump.cc [hunk 1]
```diff
@@ -46,9 +47,10 @@ class BinaryReaderObjdumpBase : public BinaryReaderNop {
  protected:
   const char* GetFunctionName(Index index);
 
-  ObjdumpOptions* options = nullptr;
-  const uint8_t* data = nullptr;
-  size_t size = 0;
+  ObjdumpOptions* options;
+  ObjdumpState* objdump_state_;
```
Comment (binji) [inline src/binary-reader-objdump.cc]: seems weird to have `objdump_state_` but the rest without trailing underscore.

Comment (binji) [inline src/binary-reader-objdump.cc]: Also, could just call this `state_`

Comment (sbc100) [inline src/binary-reader-objdump.cc]: There is already state member from the reader interface (admittedly without an _, but have state and state_ would be strange).

Comment (sbc100) [inline src/binary-reader-objdump.cc]: Removed the trailing _ for now

**Pull request merged**